monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/RecursiveConcatSuite.scala

```diff
@@ -50,4 +50,13 @@ object RecursiveConcatSuite extends BaseOperatorSuite {
     val f = range(0, count).sumL.runAsync; s.tick()
     assertEquals(f.value, Some(Success(count.toLong * (count - 1) / 2)))
   }
+
+  val nats: Observable[Long] = Observable.now(1L) ++ nats.map(_ + 1)
+
+  test("laziness of ++'s param") { implicit s =>
+    val count = 10000
+
+    import concurrent.duration._
+    assertEquals(nats.take(count).sumL.runSyncUnsafe(1.second), (1 to count).sum)
```

Comment:
Using `TestScheduler` we can manipulate time and instead of `runSyncUnsafe` use just `runAsync` and then `s.tick(1.second)` (test above) - it might also help with passing Scala.JS test :)

Reply:
I'm now trying with `tick()` following the test above's way of doing it :)